src/asciidoc/docs/blueprints/customVCC.adoc

```diff
@@ -0,0 +1,208 @@
+----
+npm install --save @withkoji/custom-vcc-sdk
+----
+
+=== Create the customization files
+Create `.koji/customization/settings.json` to define customizable values for the API URL, and the items key and item name key from the API response.
+
+[source,json]
+----
+include::../_includes/vcc-blueprint-settings.json[tags=!configuration]
+----
+<1> You will add custom values here in the next step.
+
+Remove the unused customization files.
+
+`colors.json`, `images.json`, and `strings.json`
+
+=== Configure the API
+This blueprint implements https://thecatapi.com/[The Cat API] to return a list of cat breeds.
+However, you can use any API that returns a list of items to implement a custom selector.
+
+Set the values for the API in the settings customisation file so your app can access the API.
+
+[source,json]
+----
+include::../_includes/vcc-blueprint-settings.json[tags=configuration]
+----
+
+=== Add the template logic
+Import the packages.
+
+[source,javascript]
+----
+import CustomVCC from '@withkoji/custom-vcc-sdk';
+----
+
+Initialize `CustomVCC` and set a listener to load the Koji theme.
+This listener allows you to style the VCC to match the user’s current theme on Koji (dark or light).
+
+.React
+[source,javascript]
+----
+include::../_includes/customVCCReact.js[tags=constructor,indent=0]
+----
+
+.Vanilla JS
+[source,javascript]
+----
+include::../_includes/customVCCVanilla.js[tags=initTheme,indent=0]
+----
+
+Register your custom VCC to connect to Koji and set a listener to update your application state when the custom VCC updates.
+
+.React
+[source,javascript]
+----
+include::../_includes/customVCCReact.js[tags=mount,indent=0]
+----
+.Vanilla JS
+[source,javascript]
+----
+include::../_includes/customVCCVanilla.js[tags=registerVCC,indent=0]
+----
+
+Make a call to the API endpoint to retrieve the required values.
+
+.React
+[source,javascript]
+----
+include::../_includes/customVCCReact.js[tags=loadAPI,indent=0]
+----
+
+.Vanilla JS
+[source,javascript]
+----
+include::../_includes/customVCCVanilla.js[tags=loadAPI,indent=0]
+----
+
+Create a setter to return the selected value from your application back to the VCC on Koji.
+
+Use `customVCC.change()` to notify Koji that the VCC value has been updated and `customVCC.save()` to tell Koji to commit the change to the remixer's VCC.
+
+.React
+[source,javascript]
+----
+include::../_includes/customVCCReact.js[tags=selectValue,indent=0]
+----
+.Vanilla JS
+[source,javascript]
+----
+include::../_includes/customVCCVanilla.js[tags=selectValue,indent=0]
+----
+
+Use the Koji theme and the API results to render a drop-down menu that allows users to select a value.
+
+.React
+[source,javascript]
+----
+include::../_includes/customVCCReact.js[tags=render,indent=0]
+----
+
+.Vanilla JS
+[source,javascript]
+----
+include::../_includes/customVCCVanilla.js[tags=render,indent=0]
+----
+
+=== Add the styles
+Define styles for the template, using the theme values to match the user's Koji theme.
+
```

Comment:
Luis noticed that the Koji logo is covering the selection in the example "template". Can you apply some responsive styling to adjust the vertical spacing?

Reply:
The Koji logo only shows on a published preview window, which you wouldn't have a reason to view for a VCC? The reason it is styled close to the top is so if you switch to a regular `type:select` VCC they display consistently.

Reply:
I believe it's better to have a consistent VCC input than a preview that isn't obscured. Let's leave it for now and we can circle back if it raises questions.